variables.tf

```diff
@@ -1,36 +1,164 @@
+  default     = "private"
+  validation {
+    error_message = "The specified endpoint_type can be private or public only."
+    condition     = contains(["private", "public"], var.cloud_monitoring_endpoint_type)
+  }
+}
+
+variable "cloud_monitoring_metrics_filter" {
+  type = list(object({
+    type = string
+    name = string
+  }))
+  description = "To filter custom metrics, specify the Cloud Monitoring metrics to include or to exclude. See https://cloud.ibm.com/docs/monitoring?topic=monitoring-change_kube_agent#change_kube_agent_inc_exc_metrics."
   default     = []
+  validation {
+    condition     = alltrue([for filter in var.cloud_monitoring_metrics_filter : can(regex("^(include|exclude)$", filter.type)) && filter.name != ""])
+    error_message = "The specified `type` for the `cloud_monitoring_metrics_filter` is not valid. Specify either `include` or `exclude`. The `name` field cannot be empty."
+  }
+}
+
+variable "cloud_monitoring_container_filter" {
+  type = list(object({
+    type      = string
+    parameter = string
+    name      = string
+  }))
+  description = "To filter custom containers, specify which containers to include or exclude from metrics collection for the cloud monitoring agent. See https://cloud.ibm.com/docs/monitoring?topic=monitoring-change_kube_agent#change_kube_agent_filter_data."
+  default     = []
+  validation {
+    condition     = length(var.cloud_monitoring_container_filter) == 0 || can(regex("^(include|exclude)$", var.cloud_monitoring_container_filter[0].type))
+    error_message = "Invalid input for `cloud_monitoring_container_filter`. Valid options for 'type' are: `include` and `exclude`. If empty, no containers are included or excluded."
+  }
+}
+
+variable "cloud_monitoring_agent_name" {
+  description = "Cloud Monitoring agent name. Used for naming all kubernetes and helm resources on the cluster."
+  type        = string
+  default     = "sysdig-agent"
+}
+
+variable "cloud_monitoring_agent_namespace" {
+  type        = string
+  description = "Namespace where to deploy the Cloud Monitoring agent. Default value is 'ibm-observe'"
+  default     = "ibm-observe"
+  nullable    = false
+}
+
+variable "cloud_monitoring_agent_tolerations" {
+  description = "List of tolerations to apply to Cloud Monitoring agent."
+  type = list(object({
+    key               = optional(string)
+    operator          = optional(string)
+    value             = optional(string)
+    effect            = optional(string)
+    tolerationSeconds = optional(number)
+  }))
+  default = [{
+    operator = "Exists"
+    },
+    {
+      operator = "Exists"
+      effect   = "NoSchedule"
+      key      = "node-role.kubernetes.io/master"
+  }]
+}
+
+variable "chart_location" {
+  description = "The location of the Helm chart for the Sysdig agent."
+  type        = string
+  default     = "sysdig-deploy" # Replace with the actual chart location if different
+  nullable    = false
+}
+
+variable "chart_repository" {
```

Comment:
I think maybe call this `chart_location` to be consistent with other repos. Technically this can be a local directory or a helm repo you see.